insights/specs/datasources/httpd.py

```diff
@@ -55,3 +57,67 @@ def httpd_on_nfs(broker):
             relative_path = 'insights_commands/httpd_open_nfsV4_files'
             return DatasourceProvider(content=json.dumps(result_dict), relative_path=relative_path)
     raise SkipComponent
+
+
+@datasource(HostContext)
+def httpd_configuration_files(broker):
+    """
+    This datasource returns the all of httpd configuration files' path.
+
+    Returns:
+        list: the file path of httpd configuration files
+
+    Raises:
+        SkipComponent: there is no httpd configuration file
+    """
+
+    all_paths = []
```

Comment:
@wushiqinlou - your implementation definitely works  for this issue, I just implemented it in separated functions with a recursive one, since, we should also update the `httpd_conf_scl_httpd24` and `httpd_conf_scl_jbcs_httpd24`. please have a look


```python
def _get_all_include_conf(root, glob_path):
    includes = glob_path
    # In case $ServerRoot in included in the 'glob_path'
    if not glob_path.startswith(root):
        includes = os.path.join(root, glob_path)
    # In case "*.conf" is not included in the 'glob_path'
    if not includes.endswith("*.conf"):
        includes = os.path.join(includes, "*.conf")

    _paths = set()
    try:
        for conf in glob.glob(includes):
            _paths.add(conf)
            with open(conf) as cfp:
                _includes = None
                for line in cfp.readlines()[::-1]:
                    if "Include" in line or "IncludeOptional" in line:
                        _includes = line.split()[-1].strip('"\'')
                        break
                # Search "includes" after scanning the whole file
                if _includes:
                    _paths.update(_get_all_include_conf(root, _includes))
        return _paths
    except Exception:
        pass
    return _paths

def get_httpd_configuration_files(httpd_root):
    main_httpd_conf = os.path.join(httpd_root, "conf/httpd.conf")
    all_paths = set()
    try:
        with open(main_httpd_conf) as cfp:
            server_root = httpd_root
            # Add it only when it exists
            all_paths.add(main_httpd_conf)
            for line in cfp.readlines()[::-1]:
                if "ServerRoot" in line:
                    server_root = line.strip().split()[-1].strip('"\'')
                elif "Include" in line or "IncludeOptional" in line:
                    includes = line.strip().split()[-1].strip('"\'')
            # Search "includes" after scanning the whole file
            all_paths.update(_get_all_include_conf(server_root, includes))
    except Exception:
        # Skip the datasource when no such "/etc/httpd/conf/httpd.conf" file
        raise SkipComponent

    if all_paths:
        return all_paths
    raise SkipComponent


@datasource(HostContext)
def httpd_configuration_files(broker):
    """
    This datasource returns the all of httpd configuration files' path.

    Returns:
        list: the file path of httpd configuration files

    Raises:
        SkipComponent: there is no httpd configuration file
    """
    httpd_root = '/etc/httpd'
    all_paths = get_httpd_configuration_files(httpd_root)
    if all_paths:
        return all_paths
    raise SkipComponent
```